### ISSUE TITLE
fb documentation enhancements

### ISSUE THREAD

**Issue by craigsapp** [humdrum-tools/vhv-documentation#20]
* For `--frequency`, I am thinking that `--rate` would be better (or as another long-form alias. but the primary one used in the documentation), particularly since it is shorter, and also `--frequency` is also associated with hertz in sound, which may cause confusion.
* "passend" I see is a German word.  You probably mean an English word?
* "--abbr" is good, but the primary long name should be `--abbreviate`.
* In the 9/8 examples, you use the letter `s` in measure 1 top staff?  Is that a typo, since I expect it to be an `r` as in `4.r`.  If you want in invisible rest, then use either `4.` or `4.ryy`.

Also, it would be useful to have a summary table of options.  There are not too many filters that have this yet, but a good prototype to use is the `cmr` filter
documentation:

https://doc.verovio.humdrum.org/filter/cmr/#options

<img width="743" alt="Screenshot 2023-01-22 at 8 24 31 PM" src="https://user-images.githubusercontent.com/3487289/213965694-e00cd816-9667-4a52-85e0-1f79a7532ef9.png">

This table is generated from a ATON file (similar to JSON) describing the options:

https://github.com/humdrum-tools/vhv-documentation/blob/master/filter/cmr/options.aton

Note that the question mark to the left of the table is a hyperlink to the location in the documentation where the option is discussed.  To have a hyperlink display, add a line such as:

```html
<a name="option-l"></a>
```

in the documentation where that option is discussed (`-l` in this case).

To print the options list, you add this line to `index.md`:

```liquid
{% include filter-options.html file="options.aton" lang="EN" %}
```

In the future, I am thinking of embedding this option list into the tool itself, which will generate the list when using the CLI version of the program (and perhaps in VHV somehow as well).

**Comment by WolfgangDrescher** [comment]
> For --frequency, I am thinking that --rate would be better (or as another long-form alias. but the primary one used in the documentation), particularly since it is shorter, and also --frequency is also associated with hertz in sound, which may cause confusion.

Agree, this was not the besting naming. Changed it and will open a new PR. Should I keep `--recip` as an alternative? It could be confusing since it's something different than e.g. in the `deg` filter. So maybe not.

Note that verovio on the fb documentation page will crash because of this change (at least it did on localhost). I will wait with merging this PR until humlib is updated in the verovio repo.

> "passend" I see is a German word. You probably mean an English word?

Muscle memory was typing. I meant "passed". But I rephrased it to: «This option ignores the base track passed with `-b`.»

> "--abbr" is good, but the primary long name should be --abbreviate.

Okay, this is better. I changed it.

> In the 9/8 examples, you use the letter s in measure 1 top staff? Is that a typo, since I expect it to be an r as in 4.r. If you want in invisible rest, then use either 4. or 4.ryy.

Actually no. I saw this once in a kern file somewhere and thought it stands for "silent" note some something like this. But now that you tell me I could not find any documentation about this so I guess this is not true? I changed it to `4.yy` since the rests are not displayed int my source (Henlde).

> Also, it would be useful to have a summary table of options.

Good idea, I added this.

> This table is generated from a ATON file (similar to JSON) describing the options:

JSON5 or YAML files may also be interesting for you in case you did not already know them.

> In the future, I am thinking of embedding this option list into the tool itself, which will generate the list when using the CLI version of the program (and perhaps in VHV somehow as well).

Couldn't the `define()` statements be used to generate the ATON files?

**Comment by craigsapp** [comment]
> JSON5 or YAML files may also be interesting for you in case you did not already know them.

YAML is disgusting, as is any programming language that uses leading whitespace to control syntax.

The VHV documentation sidebar is populated using YAML: https://github.com/humdrum-tools/vhv-documentation/blob/master/_data/sidebars/main_sidebar.yml

Probably I could use JSON as well in Jekyll, but I was lazy and followed the most common way of doing it.

I did not know about JSON5, but also I don't see the point as it is not sufficiently different from JSON to be of any real use.   It is basically a Javascript Object (which when I learned Javascript and JSON, I wondered why they cannot be the same, excepting function pointers and such).  So if it becomes part of the Javascript Standard and implemented in a JSON5 tool in Javascript without using an external library, I will be interested in it.

ATON has a similar usage case to JSON5: it has a javascript parser to convert to JSON.  

ATON website: https://aton.sapp.org
Github: https://github.com/craigsapp/ATON
NPM: https://www.npmjs.com/package/aton?activeTab=readme

ATON has no quotes, brackets (with exceptions for advanced usage), or commas, and minimal back-quoting, so its syntax much easier to maintain/read by hand than JSON or JSON5.   I can instantaneously explain how to create ATON files to a non-technical person, and there are very minimal caveats.

 ATON is designed to deal with large amounts of text with newlines as a data value, and most importantly, the data can be parsed without a tree-structure parser.  So in some cases where I have a very long database in this format, I can do a memoryless one-pass through the file to extract the data I want rather than loading it all into memory before starting to extract what I want.

The Tasso in music project metadata data is stored in this format:
https://github.com/TassoInMusicProject/tasso-website/tree/gh-pages/data/indexes
The longest data file being:
https://github.com/TassoInMusicProject/tasso-website/blob/gh-pages/data/indexes/rime-settings.aton

Mostly what I do now for collaborative metadata projects (particularly with non-technical musicians): I write a Google Apps Scripts interface to a Google spreadsheet to download the data according to the format I want to use, such as JSON for Javascript, or ATON for PERL, for example.

Here is an example of that:

https://docs.google.com/spreadsheets/d/1cEE6zs7u3g9mjUsr7jcUusnbp2eIkYyqsMFastC78-c/edit#gid=1130693471

That spreadsheet is downloadable as JSON, ATON, TSV and Humdrum depending on how I want to use the data:

| Data format | Link |
| --- | ---|  
| 1600s metadata, JSON format	|	https://script.google.com/macros/s/AKfycbwI-c16I3OdLH3DBysfPmBEjryrrnZEacGrLEwP_mfJupj7Vr8Kw3Pe24cZhbdCfUmalg/exec?format=json&lang=en&sheet=1600			|																					
| 1600s metadata,  ATON format	|	https://script.google.com/macros/s/AKfycbwI-c16I3OdLH3DBysfPmBEjryrrnZEacGrLEwP_mfJupj7Vr8Kw3Pe24cZhbdCfUmalg/exec?format=aton&lang=en&sheet=1600			|																					
| 1600s metadata,TSV format	|	https://script.google.com/macros/s/AKfycbwI-c16I3OdLH3DBysfPmBEjryrrnZEacGrLEwP_mfJupj7Vr8Kw3Pe24cZhbdCfUmalg/exec?format=tsv&lang=en&sheet=1600			 |																					
| 1600s metadata, Humdrum format	|	https://script.google.com/macros/s/AKfycbwI-c16I3OdLH3DBysfPmBEjryrrnZEacGrLEwP_mfJupj7Vr8Kw3Pe24cZhbdCfUmalg/exec?format=humdrum&lang=en&sheet=1600	 |																							

> Couldn't the define() statements be used to generate the ATON files?

Yes, that is the vague idea.  And in addition there could be formatting control to output on JSON, JSON5, YML

https://xkcd.com/927

**Comment by craigsapp** [comment]
> Actually no. I saw this once in a kern file somewhere and thought it stands for "silent" note some something like this. But now that you tell me I could not find any documentation about this so I guess this is not true? I changed it to 4.yy since the rests are not displayed int my source (Henlde).

No, it has never meant silent.  It could be an error in another data file.  When you use something like `4.s` the Humdrum-to-MEI parser cannot understand the `s`, so it is ignored, and the token is treated as `4.` which gets converted as an implicit invisible rest (called a `<space>` in MEI).  `4.ryy` is an explicitly invisible rests.   

Currently `4.` and `4.ryy` are treated the same in the Humdrum-to-MEI converter, but in the future I am thinking of using them in slightly different ways:  the implicit rests would be useful to have a conversion from spine multiple spine splits/mergers as often seen in piano music -- into a single split at the start and a merge at the bottom.  For cases where the original file merged, there will be an extra subspine filled in with implicit rests.   This would allow converting back and forth between the two forms of spining (verbose and collapsed).   Explicit rests such as `4ryy` would not be collapsible, but `4.` would.

**Comment by WolfgangDrescher** [comment]
> YAML is disgusting, as is any programming language that uses leading whitespace to control syntax.

This is why I avoid python 🤣️. But for small configuration files YAML is fine for me.

> The Tasso in music project metadata data is stored in this format

This is interesting. I'm doing something similar in https://github.com/WolfgangDrescher/lassus-tricinium-project/ but with YAML.
